### PR TITLE
Add csv-stringify

### PR DESCRIPTION
**Pull request by IegorT**
- [x] code is properly formatted (`npm run fmt`)
- [ ] tests are added/updated
- [ ] documentation is updated (`npm run doc` to regenerate documentation based on comments)
- [ ] description of changes is added under the `Unreleased` header in CHANGELOG.md

### CODE REVIEW

File: lib/csv.js [hunk 1]
```diff
@@ -1,7 +1,336 @@
...
 
-module.exports = { stringify };
+function getPlatformDelimiter() {
+  const opSys = process.platform;
+  const systems = {
+    win32: recordDelimiters.windows,
+    darwin: recordDelimiters.mac,
+    linux: recordDelimiters.unix,
+  };
+
```
[the remark this answers is not in the document]
Comment (IegorT) [reply lib/csv.js]: Fixed

File: lib/csv.js [hunk 2]
```diff
@@ -1,7 +1,336 @@
...
+    const options =
+      args.length === 3 ? Options.create(args[1]) : Options.create();
+    if (options.columns === null) options.addHeader(data[0]);
+    const result = createCSV(data, options);
+    return process.nextTick(callback, null, result);
+  } catch (error) {
+    return process.nextTick(callback, error);
+  }
+}
```
[the remark this answers is not in the document]
Comment (IegorT) [reply lib/csv.js]: Accepted and changed

File: .gitignore [hunk 3]
```diff
@@ -3,4 +3,4 @@ node_modules
 .DS_Store
 .nyc_output
 coverage
-dist
+dist
```
Comment (yehorbk) [inline .gitignore]:
```suggestion
dist

```

File: lib/csv.js [hunk 4]
```diff
@@ -125,26 +128,59 @@ function checkValue(value, type, options) {
...
+      value = quote + value + quote;
+    }
+  } else if (
+    options.quotedEmpty === true ||
+    (!options.quotedEmpty && value === '' && options.quotedString !== false)
+  ) {
+    const quote = options.quote;
+    value += quote + quote;
+  }
+
```
Comment (yehorbk) [inline lib/csv.js]:
```suggestion
```

**Comment by belochub** [comment]
@yehorbk @IegorT are you planning on finishing your work on this PR? If not, I'm going to close it.

**Pull request closed without merging**